ledger/src/shred.rs

```diff
@@ -780,6 +787,45 @@ pub mod layout {
         Ok(())
     }
 
+    /// Resigns the shred's Merkle root as the retransmitter node in the
+    /// Turbine broadcast tree. This signature is in addition to leader's
+    /// signature which is left intact.
+    pub fn resign_shred(shred: &mut [u8], keypair: &Keypair) -> Result<(), Error> {
+        let (offset, merkle_root) = match get_shred_variant(shred)? {
+            ShredVariant::LegacyCode | ShredVariant::LegacyData => {
+                return Err(Error::InvalidShredVariant)
+            }
+            ShredVariant::MerkleCode {
+                proof_size,
+                chained,
+                resigned,
```

Comment:
We expect resigned to be true here, right? Is the idea that we will catch this in `get_retransmitter_signature_offset` and return appropriate Error if that assumption is violated?

Reply:
yes.
Ideally `shred::layout::*` implement minimal logic and only do the dispatching to the correct submodule.
This style also avoids the kind of problem like the one address in https://github.com/anza-xyz/agave/pull/802. I.e. `get_retransmitter_signature_offset` should enforce right variant anyways, which makes it unnecessary to check here.